pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -147,6 +152,22 @@ func newAlertmanagerMetrics() *alertmanagerMetrics {
 			"cortex_alertmanager_config_hash",
 			"Hash of the currently loaded alertmanager configuration.",
 			[]string{"user"}, nil),
+		partialMerges: prometheus.NewDesc(
+			"cortex_alertmanager_partial_state_merges_total",
+			"Number of times we have received a partial state to merge for a key.",
+			[]string{"key"}, nil),
```

Comment:
We don't have user as part of these labels because `key` already includes the user.  It's a combination of `prefix + userID`. So by using key we "technically get both". Even though it breaks the nomenclature (of always using `user`).

Reply:
We have per-user registries, so we could do per-user output, aggregating over all keys. WDYT?

Reply:
Is that different from the current approach? An aggregation across all keys is not possible if the keys are per user e.g. `sil:user-3` or `nfl:user-2`.

Reply:
If I understand it correctly, we have per-user registries, which only use "key" label. During `alertmanagerMetrics.Collect`, when then call `SendSumOfCountersWithLabels` with `key` label. We could instead call `SendSumOfCountersPerUser` (eg. `data.SendSumOfCountersPerUser(out, m.partialMerges, "alertmanager_partial_state_merges_total")`), which would return `sum(alertmanager_partial_state_merges_total)` per user-registry, and then add `user` label to the output. I think that would be enough. WDYT?